pkg/apis/duck/v1alpha1/credentials.go

```diff
@@ -30,37 +30,24 @@ const (
 )
 
 var (
-	// The format of gServiceAccountName is service-account-name@project-id.iam.gserviceaccount.com
-	// Service account name must be between 6 and 30 characters (inclusive),
-	// must begin with a lowercase letter, and consist of lowercase alphanumeric characters that can be separated by hyphens.
-	// Project IDs must start with a lowercase letter and can have lowercase ASCII letters, digits or hyphens,
-	// must be between 6 and 30 characters.
-	gsaValidationRegex = regexp.MustCompile(`^[a-z][a-z0-9-]{5,29}@[a-z][a-z0-9-]{5,29}.iam.gserviceaccount.com$`)
-
 	// The name of a k8s ServiceAccount object must be a valid DNS subdomain name.
 	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
 	ksaValidationRegex = regexp.MustCompile(`^[A-Za-z0-9](?:[A-Za-z0-9\-]{0,61}[A-Za-z0-9])?$`)
 )
 
 // ValidateCredential checks secret and service account.
-func ValidateCredential(secret *corev1.SecretKeySelector, gServiceAccountName string, kServiceAccountName string) *apis.FieldError {
+func ValidateCredential(secret *corev1.SecretKeySelector, kServiceAccountName string) *apis.FieldError {
 	if secret != nil && !equality.Semantic.DeepEqual(secret, &corev1.SecretKeySelector{}) && kServiceAccountName != "" {
 		return &apis.FieldError{
 			Message: "Can't have spec.serviceAccountName and spec.secret at the same time",
 			Paths:   []string{""},
 		}
 	} else if secret != nil && !equality.Semantic.DeepEqual(secret, &corev1.SecretKeySelector{}) {
 		return validateSecret(secret)
-	} else {
-		var errs *apis.FieldError
-		if kServiceAccountName != "" {
-			errs = errs.Also(validateK8sServiceAccount(kServiceAccountName))
-		}
-		if gServiceAccountName != "" {
-			errs = errs.Also(validateGCPServiceAccount(gServiceAccountName))
-		}
-		return errs
+	} else if kServiceAccountName != "" {
```

Comment:
Is it possible that both secret and kServiceAccountName is nil or "", do we allow that?

Reply:
We allow that case, defaulting webhook will add default credential to it. (which is set up in configmap config-gcp-auth).